### PR TITLE
[RPO] fix nll loss

### PR DESCRIPTION
**Pull request by kashif**
fixes the nll for RPO loss from #1686 and log it when `rpo_alpha` is not `None`

### CODE REVIEW

**Comment by kashif** [comment]
@winglian can you kindly try with this fix?

File: trl/trainer/dpo_trainer.py [hunk 1]
```diff
@@ -1225,7 +1241,7 @@ def get_batch_loss_metrics(
         reward_accuracies = (chosen_rewards > rejected_rewards).float()
 
         if self.args.rpo_alpha is not None:
-            losses = losses * self.args.rpo_alpha - policy_chosen_logps_avg
+            losses = losses * self.args.rpo_alpha + policy_nll_loss
```
Comment (lewtun) [inline trl/trainer/dpo_trainer.py:1244]: Perhaps I'm missing something, but isn't the NLL loss normalized by the total sequence length? 


Screenshot 2024-06-06 at 14 54 39

**Review by lewtun — COMMENTED** [review]
Code LGTM, but I have one question about where we normalize the NLL loss

File: trl/trainer/dpo_trainer.py (hunk 1 above)
Comment (kashif) [reply trl/trainer/dpo_trainer.py:1244]: yes the `CrossEntropyLoss` is reducing by `mean` by default

**Comment by kashif** [comment]
also note @lewtun the v2 of the paper has this loss averaged over the completions (c and y):

Screenshot 2024-06-06 at 17 47 49

**Review by lewtun — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**